name: Lint, Build and scan container image

on:
  pull_request:

jobs:
  lint:
    name: Lint Dockerfile with hadolint
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
      - uses: hadolint/hadolint-action@2332a7b74a6de0dda2e2221d575162eba76ba5e5 # v3.3.0
        with:
          dockerfile: Dockerfile
          failure-threshold: error
          verbose: true

  build:
    name: Build and scan container image
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0

      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@e468171a9de216ec08956ac3ada2f0791b6bd435 # v3.11.1

      - name: Build only
        uses: docker/build-push-action@263435318d21b8e681c14492fe198d362a7d2c83 # v6.18.0
        with:
          context: .
          file: ./Dockerfile
          load: true
          push: false
          tags: diag-tools:dev

      - name: Dockle
        uses: erzz/dockle-action@69369bc745ee29813f730231a821bcd4f71cd290 # v1.4.0
        with:
          image: diag-tools:dev
          exit-code: '1'
          failure-threshold: fatal
          # google-cloud-sdk の settings.py が CIS-DI-0010 (FATAL) に引っかかるので除外
          accept-filenames: settings.py

      - name: Trivy
        uses: aquasecurity/trivy-action@b6643a29fecd7f34b3597bc6acb0a98b03d33ff8 # 0.33.1
        with:
          image-ref: diag-tools:dev
          format: table
          exit-code: '1'
          ignore-unfixed: true
          vuln-type: 'os,library'
          severity: 'CRITICAL,HIGH'
